WhatsApp/app/src/main/res/layout/fragment_notifications.xml

```diff
@@ -1,6 +1,564 @@
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_marginStart="16dp"
+                android:layout_marginBottom="4dp"
+                android:text="@string/notificationtone" />
+
+            <TextView
+                android:id="@+id/notificationToneDesc"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_below="@+id/tvnotificationTones"
+                android:layout_marginStart="16dp"
+                android:text="Default ringtone (Bongo)" />
+
+
+
+        </RelativeLayout>
+
+        <RelativeLayout
+            android:id="@+id/vibrateLayout"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:clickable="true"
+            android:focusable="true"
+            android:paddingTop="12dp"
+            android:paddingBottom="12dp"
+            android:background="?attr/selectableItemBackground">
+
+
+            <TextView
+                android:id="@+id/vibrateTones"
+                style="@style/NotificationsTitleText"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_marginStart="16dp"
+                android:layout_marginBottom="4dp"
+
+                android:text="@string/vibrate" />
+
+            <TextView
+                android:id="@+id/vibrateDesc"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_below="@+id/vibrateTones"
+                android:layout_marginStart="16dp"
+                android:text="Default ringtone (Bongo)" />
+
+
+
+        </RelativeLayout>
+
+        <RelativeLayout
+            android:id="@+id/popUpLayout"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:clickable="true"
+            android:focusable="true"
+            android:paddingTop="12dp"
+            android:paddingBottom="12dp"
+            android:background="?attr/selectableItemBackground">
+
+
+            <TextView
+                android:id="@+id/tvPopUpTones"
+                style="@style/NotificationsTitleText"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_marginStart="16dp"
+                android:layout_marginBottom="4dp"
+                android:text="@string/popupnotification" />
+
+            <TextView
+                android:id="@+id/tvPopUpTonesDesc"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_below="@+id/tvPopUpTones"
+                android:layout_marginStart="16dp"
+                android:text="@string/nopopup" />
```

Comment:
I believe these text views should be initialized in the fragment and the text should be obtained through some sort of local cache like shared preferences or data store. We don't want to hard code the default here.

Reply:
its just here until we add the features and im going to take it out